### ISSUE TITLE
No storage offers were received

### ISSUE THREAD

**Issue by 1credit** [Storj/bridge#216]
Package Versions

Replace the values below using the output from npm list storj. Use npm list -g storj if installed globally.

Storj 3.1.2
Replace the values below using the output from node --version.

v4.4.7
Expected Behavior

Network would not fail with lack of storage offers given the amount of unused space on the net.

Actual Behavior

```
dd5dcf0322450c9620077e33c2f3b5095e2f5 (retry: 2)
[Wed Sep 07 2016 21:42:31 GMT-0500 (CDT)] [info] Querying bridge for contract for 50f 3f4d77ff3fcb32c0dc59f7dcf048563d07112 (retry: 18)
[Wed Sep 07 2016 21:43:00 GMT-0500 (CDT)] [info] Querying bridge for contract for 2df dd5dcf0322450c9620077e33c2f3b5095e2f5 (retry: 3)
[Wed Sep 07 2016 21:43:03 GMT-0500 (CDT)] [info] Contract negotiated with: {"protocol ":"0.8.1","address":"jsia.ddns.net","port":23550,"nodeID":"29dff6268e24afe024ddcae4e574f 41d4af23edb","lastSeen":1473302582091}
[Wed Sep 07 2016 21:43:03 GMT-0500 (CDT)] [info] Data channel opened, transferring sh ard...
[Wed Sep 07 2016 21:43:17 GMT-0500 (CDT)] [info] Querying bridge for contract for 50f 3f4d77ff3fcb32c0dc59f7dcf048563d07112 (retry: 19)
[Wed Sep 07 2016 21:44:01 GMT-0500 (CDT)] [info] Shard transfer completed! 9 remainin g...
[Wed Sep 07 2016 21:44:01 GMT-0500 (CDT)] [info] Trying to upload shard /tmp/00fe6600 d61c index 32
[Wed Sep 07 2016 21:44:01 GMT-0500 (CDT)] [info] Hash for this shard is: 1c534310ea34 74b6e8226d842871a61941018425
[Wed Sep 07 2016 21:44:03 GMT-0500 (CDT)] [info] Audit generation for shard done.
[Wed Sep 07 2016 21:44:03 GMT-0500 (CDT)] [info] Waiting on a storage offer from the network...
[Wed Sep 07 2016 21:44:03 GMT-0500 (CDT)] [info] Querying bridge for contract for 1c5 34310ea3474b6e8226d842871a61941018425 (retry: 0)
[Wed Sep 07 2016 21:44:10 GMT-0500 (CDT)] [info] Querying bridge for contract for 50f 3f4d77ff3fcb32c0dc59f7dcf048563d07112 (retry: 20)
[Wed Sep 07 2016 21:44:19 GMT-0500 (CDT)] [info] Querying bridge for contract for 1c5 34310ea3474b6e8226d842871a61941018425 (retry: 1)
[Wed Sep 07 2016 21:44:20 GMT-0500 (CDT)] [info] Contract negotiated with: {"protocol ":"0.8.0","address":"83.248.66.115","port":4000,"nodeID":"175f699bbf05bcc0a5e4a1674096ac f836d14c35","lastSeen":1473302660059}
[Wed Sep 07 2016 21:44:21 GMT-0500 (CDT)] [info] Data channel opened, transferring sh ard...
[Wed Sep 07 2016 21:44:27 GMT-0500 (CDT)] [info] Querying bridge for contract for 50f 3f4d77ff3fcb32c0dc59f7dcf048563d07112 (retry: 21)
[Wed Sep 07 2016 21:44:43 GMT-0500 (CDT)] [info] Querying bridge for contract for 50f 3f4d77ff3fcb32c0dc59f7dcf048563d07112 (retry: 22)
[Wed Sep 07 2016 21:45:08 GMT-0500 (CDT)] [info] Shard transfer completed! 8 remainin g...
[Wed Sep 07 2016 21:45:08 GMT-0500 (CDT)] [info] Trying to upload shard /tmp/dcfb4727 a182 index 33
[Wed Sep 07 2016 21:45:08 GMT-0500 (CDT)] [info] Hash for this shard is: ec550b8bd630 9345d6b3abb611a16b09b1fa902e
[Wed Sep 07 2016 21:45:11 GMT-0500 (CDT)] [info] Audit generation for shard done.
[Wed Sep 07 2016 21:45:11 GMT-0500 (CDT)] [info] Waiting on a storage offer from the network...
[Wed Sep 07 2016 21:45:11 GMT-0500 (CDT)] [info] Querying bridge for contract for ec5 50b8bd6309345d6b3abb611a16b09b1fa902e (retry: 0)
[Wed Sep 07 2016 21:45:21 GMT-0500 (CDT)] [info] Querying bridge for contract for 50f 3f4d77ff3fcb32c0dc59f7dcf048563d07112 (retry: 23)
[Wed Sep 07 2016 21:45:26 GMT-0500 (CDT)] [info] Querying bridge for contract for ec5 50b8bd6309345d6b3abb611a16b09b1fa902e (retry: 1)
[Wed Sep 07 2016 21:45:37 GMT-0500 (CDT)] [info] Querying bridge for contract for 50f 3f4d77ff3fcb32c0dc59f7dcf048563d07112 (retry: 24)
[Wed Sep 07 2016 21:45:41 GMT-0500 (CDT)] [info] Querying bridge for contract for ec5 50b8bd6309345d6b3abb611a16b09b1fa902e (retry: 2)
[Wed Sep 07 2016 21:45:53 GMT-0500 (CDT)] [warn] Failed to upload shard...
[Wed Sep 07 2016 21:45:53 GMT-0500 (CDT)] [warn] [ fspoe.tar ] Error occurred. Trigge ring cleanup...
[Wed Sep 07 2016 21:45:53 GMT-0500 (CDT)] [info] [ fspoe.tar ] Cleaning up...
[Wed Sep 07 2016 21:45:54 GMT-0500 (CDT)] [info] [ fspoe.tar ] Finished cleaning!
[Wed Sep 07 2016 21:45:54 GMT-0500 (CDT)] [error] [ undefined ] A file has failed to u pload: Error: No storage offers were received
```

Steps to Reproduce

Please include the steps the reproduce the issue, numbered below. Include as
much detail as possible.

...
...
...
 @littleskunk
 
littleskunk commented 15 hours ago • edited
This is the renter view of Storj/storjshare-cli#122 (comment)

The bridge would like to send the shard 50f3f4d77ff3fcb32c0dc59f7dcf048563d07112 to his own neighbors 593844dc7f0076a1aeda9a6b9788af17e67c1052.

At the same time the bridge will relay contract PUBLISH messages of at least 13 other bridge server. Neighbors are unable to relay the messages, unable to send OFFER messages and at the end they will timeout.

At the end one more issue for #221

**Comment by littleskunk** [comment]
Update:

Last week the traffic was sharded between 14 bridge server. My own node was running with a whitelist (api.storj.io only) at ~25% CPU usage. I am close to one of the bridge server and received more PUBLISH messages than I should receive but I was able to relay them in time.

Now all 13 storj.dk server are sending contracts with the same renterID. Lets take a simple example to explain the result:
Lets take 5 bridge server (1 storj.io and 4 storj.dk) and 5 farmer. Each bridge send out one contract PUBLISH. Farmer will start locating the renter. Last week all 5 farmer will locate all 5 bridge server. This week they will only locate 2 bridge server because 4 of them send out the same renterID. 
The result: All 5 nodes will relay complete traffic to storj.io and one of the 4 bridge server. More traffic for storj.io and neigbors.
Even with my whitelist my CPU is at 100% now and I am not close to api.storj.io.

On the other side I noticed that my whitelist is sending a few OFFER to storj.io. Last week I get a lot of "to late" messages but this time I get at least 90% of the shards. That looks like the bridge is receiving only a few OFFER now.

**Comment by littleskunk** [comment]
And I think the 13 storj.dk server are sending more messages as well but I can't proof it. I receive a PUBLISH message from server 19 and send an OFFER to server 20. What will server 20 do? He can't upload the shard right? On the other side server19 did not receive any OFFER message? He has to retry it? -> More PUBLISH traffic than last week.

**Comment by MeijeSibbel** [comment]
[Fri Sep 09 2016 12:05:48 GMT+0200 (W. Europe Daylight Time)] [info] Querying bridge for contract for 211d9e09cfcdabb9560a093816afe2ca14eea8df (retry: 0)
[Fri Sep 09 2016 12:05:52 GMT+0200 (W. Europe Daylight Time)] [info] Querying bridge for contract for 3a452907fd256a5b6ec45bef2141c32ec3724723 (retry: 10)
[Fri Sep 09 2016 12:05:55 GMT+0200 (W. Europe Daylight Time)] [info] Contract negotiated with: {"protocol":"0.8.1","address":"84.136.127.167","port":26879,
odeID":"3c6d0b92a944bb4a3786944db75cdaa04e6c74eb","lastSeen":1473415548307}
[Fri Sep 09 2016 12:05:56 GMT+0200 (W. Europe Daylight Time)] [info] Data channel opened, transferring shard...
[Fri Sep 09 2016 12:05:56 GMT+0200 (W. Europe Daylight Time)] [info] Querying bridge for contract for 47b5f8b1b27699685cd55312a5ad995984b30c60 (retry: 15)
[Fri Sep 09 2016 12:05:58 GMT+0200 (W. Europe Daylight Time)] [info] Contract negotiated with: {"protocol":"0.8.1","address":"redcircle.ddns.net","port":40
,"nodeID":"eae9095e8af3ea1fa1c1c519225baf5c94672e25","lastSeen":1473415543805}
[Fri Sep 09 2016 12:06:12 GMT+0200 (W. Europe Daylight Time)] [info] Querying bridge for contract for 47b5f8b1b27699685cd55312a5ad995984b30c60 (retry: 16)
[Fri Sep 09 2016 12:06:19 GMT+0200 (W. Europe Daylight Time)] [warn] Failed to transfer shard, reason: connect ETIMEDOUT 73.40.252.127:4000
[Fri Sep 09 2016 12:06:19 GMT+0200 (W. Europe Daylight Time)] [info] Retrying shard transfer, pointer: {"hash":"211d9e09cfcdabb9560a093816afe2ca14eea8df","
ken":"3692beee496b90da3739de750b583f65ec911f12","operation":"PUSH","farmer":{"protocol":"0.8.1","address":"redcircle.ddns.net","port":4000,"nodeID":"eae9095e8
3ea1fa1c1c519225baf5c94672e25","lastSeen":1473415543805}}
[Fri Sep 09 2016 12:06:23 GMT+0200 (W. Europe Daylight Time)] [info] Querying bridge for contract for 95e5d24965bc289e07227415f8c0ff84652b481a (retry: 3)
[Fri Sep 09 2016 12:06:32 GMT+0200 (W. Europe Daylight Time)] [info] Contract negotiated with: {"protocol":"0.8.1","address":"91.244.113.103","port":5000,"
deID":"9d0c8cc0f05d4c073cc00d209cf4983df955b824","lastSeen":1473415582600}
[Fri Sep 09 2016 12:06:35 GMT+0200 (W. Europe Daylight Time)] [info] Data channel opened, transferring shard...
[Fri Sep 09 2016 12:06:38 GMT+0200 (W. Europe Daylight Time)] [info] Querying bridge for contract for 47b5f8b1b27699685cd55312a5ad995984b30c60 (retry: 17)
[Fri Sep 09 2016 12:06:40 GMT+0200 (W. Europe Daylight Time)] [warn] Failed to transfer shard, reason: connect ETIMEDOUT 73.40.252.127:4000
[Fri Sep 09 2016 12:06:40 GMT+0200 (W. Europe Daylight Time)] [info] Retrying shard transfer, pointer: {"hash":"211d9e09cfcdabb9560a093816afe2ca14eea8df","
ken":"3692beee496b90da3739de750b583f65ec911f12","operation":"PUSH","farmer":{"protocol":"0.8.1","address":"redcircle.ddns.net","port":4000,"nodeID":"eae9095e8
3ea1fa1c1c519225baf5c94672e25","lastSeen":1473415543805}}
[Fri Sep 09 2016 12:06:43 GMT+0200 (W. Europe Daylight Time)] [info] Shard transfer completed! 27 remaining...
[Fri Sep 09 2016 12:06:43 GMT+0200 (W. Europe Daylight Time)] [info] Trying to upload shard D:\temp\8d0b4d52e8f0 index 26
[Fri Sep 09 2016 12:06:44 GMT+0200 (W. Europe Daylight Time)] [info] Hash for this shard is: cb370a2e9067d1014ff8ea035a570e86ad2455c1
[Fri Sep 09 2016 12:06:46 GMT+0200 (W. Europe Daylight Time)] [info] Audit generation for shard done.
[Fri Sep 09 2016 12:06:46 GMT+0200 (W. Europe Daylight Time)] [info] Waiting on a storage offer from the network...
[Fri Sep 09 2016 12:06:46 GMT+0200 (W. Europe Daylight Time)] [info] Querying bridge for contract for cb370a2e9067d1014ff8ea035a570e86ad2455c1 (retry: 0)
[Fri Sep 09 2016 12:06:54 GMT+0200 (W. Europe Daylight Time)] [info] Querying bridge for contract for 47b5f8b1b27699685cd55312a5ad995984b30c60 (retry: 18)
[Fri Sep 09 2016 12:07:01 GMT+0200 (W. Europe Daylight Time)] [warn] Failed to transfer shard, reason: connect ETIMEDOUT 73.40.252.127:4000
[Fri Sep 09 2016 12:07:01 GMT+0200 (W. Europe Daylight Time)] [info] Retrying shard transfer, pointer: {"hash":"211d9e09cfcdabb9560a093816afe2ca14eea8df","
ken":"3692beee496b90da3739de750b583f65ec911f12","operation":"PUSH","farmer":{"protocol":"0.8.1","address":"redcircle.ddns.net","port":4000,"nodeID":"eae9095e8
3ea1fa1c1c519225baf5c94672e25","lastSeen":1473415543805}}
[Fri Sep 09 2016 12:07:04 GMT+0200 (W. Europe Daylight Time)] [info] Querying bridge for contract for cb370a2e9067d1014ff8ea035a570e86ad2455c1 (retry: 1)
[Fri Sep 09 2016 12:07:10 GMT+0200 (W. Europe Daylight Time)] [info] Querying bridge for contract for 47b5f8b1b27699685cd55312a5ad995984b30c60 (retry: 19)
[Fri Sep 09 2016 12:07:23 GMT+0200 (W. Europe Daylight Time)] [warn] Failed to transfer shard, reason: connect ETIMEDOUT 73.40.252.127:4000
[Fri Sep 09 2016 12:07:23 GMT+0200 (W. Europe Daylight Time)] [info] Shard transfer failed 3 times, getting another contract...
[Fri Sep 09 2016 12:07:23 GMT+0200 (W. Europe Daylight Time)] [info] Hash for this shard is: 211d9e09cfcdabb9560a093816afe2ca14eea8df
[Fri Sep 09 2016 12:07:25 GMT+0200 (W. Europe Daylight Time)] [info] Audit generation for shard done.
[Fri Sep 09 2016 12:07:25 GMT+0200 (W. Europe Daylight Time)] [info] Waiting on a storage offer from the network...
[Fri Sep 09 2016 12:07:25 GMT+0200 (W. Europe Daylight Time)] [info] Querying bridge for contract for 211d9e09cfcdabb9560a093816afe2ca14eea8df (retry: 0)
[Fri Sep 09 2016 12:07:25 GMT+0200 (W. Europe Daylight Time)] [info] Querying bridge for contract for 47b5f8b1b27699685cd55312a5ad995984b30c60 (retry: 20)
[Fri Sep 09 2016 12:07:30 GMT+0200 (W. Europe Daylight Time)] [info] Shard transfer completed! 26 remaining...
[Fri Sep 09 2016 12:07:30 GMT+0200 (W. Europe Daylight Time)] [info] Trying to upload shard D:\temp\2c6178e914b3 index 27
[Fri Sep 09 2016 12:07:31 GMT+0200 (W. Europe Daylight Time)] [info] Hash for this shard is: 7ddc08c618739446a912136cc8c2d679b14d59bb
[Fri Sep 09 2016 12:07:31 GMT+0200 (W. Europe Daylight Time)] [info] Contract negotiated with: {"protocol":"0.8.1","address":"78.69.217.113","port":37309,"
deID":"cdd94dcaa262ad3e9285960ad677299d4810d717","lastSeen":1473415624946}
[Fri Sep 09 2016 12:07:32 GMT+0200 (W. Europe Daylight Time)] [info] Data channel opened, transferring shard...
[Fri Sep 09 2016 12:07:32 GMT+0200 (W. Europe Daylight Time)] [info] Audit generation for shard done.
[Fri Sep 09 2016 12:07:32 GMT+0200 (W. Europe Daylight Time)] [info] Waiting on a storage offer from the network...
[Fri Sep 09 2016 12:07:32 GMT+0200 (W. Europe Daylight Time)] [info] Querying bridge for contract for 7ddc08c618739446a912136cc8c2d679b14d59bb (retry: 0)
[Fri Sep 09 2016 12:07:42 GMT+0200 (W. Europe Daylight Time)] [info] Querying bridge for contract for 47b5f8b1b27699685cd55312a5ad995984b30c60 (retry: 21)
[Fri Sep 09 2016 12:07:48 GMT+0200 (W. Europe Daylight Time)] [info] Querying bridge for contract for 7ddc08c618739446a912136cc8c2d679b14d59bb (retry: 1)
[Fri Sep 09 2016 12:07:48 GMT+0200 (W. Europe Daylight Time)] [info] Querying bridge for contract for 211d9e09cfcdabb9560a093816afe2ca14eea8df (retry: 1)
[Fri Sep 09 2016 12:07:52 GMT+0200 (W. Europe Daylight Time)] [info] Contract negotiated with: {"protocol":"0.8.1","address":"010s.storjlt.xyz","port":4000
nodeID":"7821b4357e9fd8c0e2a08a9ecc80fca70e8e9a8e","lastSeen":1473415665030}
[Fri Sep 09 2016 12:07:52 GMT+0200 (W. Europe Daylight Time)] [info] Data channel opened, transferring shard...
[Fri Sep 09 2016 12:07:53 GMT+0200 (W. Europe Daylight Time)] [info] Shard transfer completed! 25 remaining...
[Fri Sep 09 2016 12:07:53 GMT+0200 (W. Europe Daylight Time)] [info] Trying to upload shard D:\temp\6b5224f6638c index 28
[Fri Sep 09 2016 12:07:54 GMT+0200 (W. Europe Daylight Time)] [info] Hash for this shard is: 64dc13c98992a902cb87ef468864696283563040
[Fri Sep 09 2016 12:07:55 GMT+0200 (W. Europe Daylight Time)] [info] Audit generation for shard done.
[Fri Sep 09 2016 12:07:55 GMT+0200 (W. Europe Daylight Time)] [info] Waiting on a storage offer from the network...
[Fri Sep 09 2016 12:07:55 GMT+0200 (W. Europe Daylight Time)] [info] Querying bridge for contract for 64dc13c98992a902cb87ef468864696283563040 (retry: 0)
[Fri Sep 09 2016 12:07:58 GMT+0200 (W. Europe Daylight Time)] [info] Querying bridge for contract for 47b5f8b1b27699685cd55312a5ad995984b30c60 (retry: 22)
[Fri Sep 09 2016 12:08:07 GMT+0200 (W. Europe Daylight Time)] [info] Shard transfer completed! 24 remaining...
[Fri Sep 09 2016 12:08:07 GMT+0200 (W. Europe Daylight Time)] [info] Trying to upload shard D:\temp\688b3003b1c8 index 29
[Fri Sep 09 2016 12:08:08 GMT+0200 (W. Europe Daylight Time)] [info] Hash for this shard is: b8c713f6530b33e3290e00cc83375d92acd10478
[Fri Sep 09 2016 12:08:09 GMT+0200 (W. Europe Daylight Time)] [info] Audit generation for shard done.
[Fri Sep 09 2016 12:08:09 GMT+0200 (W. Europe Daylight Time)] [info] Waiting on a storage offer from the network...
[Fri Sep 09 2016 12:08:09 GMT+0200 (W. Europe Daylight Time)] [info] Querying bridge for contract for b8c713f6530b33e3290e00cc83375d92acd10478 (retry: 0)
[Fri Sep 09 2016 12:08:11 GMT+0200 (W. Europe Daylight Time)] [info] Querying bridge for contract for 64dc13c98992a902cb87ef468864696283563040 (retry: 1)
[Fri Sep 09 2016 12:08:14 GMT+0200 (W. Europe Daylight Time)] [info] Querying bridge for contract for 47b5f8b1b27699685cd55312a5ad995984b30c60 (retry: 23)
[Fri Sep 09 2016 12:08:27 GMT+0200 (W. Europe Daylight Time)] [info] Querying bridge for contract for 64dc13c98992a902cb87ef468864696283563040 (retry: 2)
[Fri Sep 09 2016 12:08:27 GMT+0200 (W. Europe Daylight Time)] [info] Querying bridge for contract for 211d9e09cfcdabb9560a093816afe2ca14eea8df (retry: 2)
[Fri Sep 09 2016 12:08:30 GMT+0200 (W. Europe Daylight Time)] [info] Querying bridge for contract for 47b5f8b1b27699685cd55312a5ad995984b30c60 (retry: 24)
[Fri Sep 09 2016 12:08:42 GMT+0200 (W. Europe Daylight Time)] [info] Querying bridge for contract for 64dc13c98992a902cb87ef468864696283563040 (retry: 3)
[Fri Sep 09 2016 12:08:43 GMT+0200 (W. Europe Daylight Time)] [info] Querying bridge for contract for 211d9e09cfcdabb9560a093816afe2ca14eea8df (retry: 3)
[Fri Sep 09 2016 12:08:45 GMT+0200 (W. Europe Daylight Time)] [warn] Failed to upload shard...
[Fri Sep 09 2016 12:08:45 GMT+0200 (W. Europe Daylight Time)] [warn] [ A.Promise.2013.720p.BluRay.x264.YIFY.mp4 ] Error occurred. Triggering cleanup...
[Fri Sep 09 2016 12:08:45 GMT+0200 (W. Europe Daylight Time)] [info] [ A.Promise.2013.720p.BluRay.x264.YIFY.mp4 ] Cleaning up...
[Fri Sep 09 2016 12:08:45 GMT+0200 (W. Europe Daylight Time)] [info] [ A.Promise.2013.720p.BluRay.x264.YIFY.mp4 ] Finished cleaning!
[Fri Sep 09 2016 12:08:45 GMT+0200 (W. Europe Daylight Time)] [error] [ undefined ] A file has failed to upload: Error: No storage offers were received

C:\Windows\system32>

**Comment by MeijeSibbel** [comment]
Sometimes this occurs due to internet connection glitches.

**Comment by super3** [comment]
@1credit please use ``` so we can actually read your logs.

**Comment by Th3Van** [comment]
@MeijeSibbel - It would be interesting to know if you would be getting the same error, trying to upload the same file using one of the storj.dk servers. Fell free to register and try uploading as many files as you like. 
The server range are : `server012.storj.dk` -> `server025.storj.dk` (just pick a random one).
Remember to use the -u option like this on every CLI command : "storj -u http://server015.storj.dk register"

The upload script on all fourteen servers are currently on hold, so you should be able to plenty of offers :)

**Comment by MeijeSibbel** [comment]
@Th3Van , if i try that i get:

C:\Users\Dani Marion>storj -k meije0612 -u http://server015.storj.dk add-bucket Test 1000 1000
 [Sat Sep 10 2016 15:47:35 GMT+0200 (W. Europe Daylight Time)] [error]  Public key not registered

C:\Users\Dani Marion>

**Comment by MeijeSibbel** [comment]
That does not work:

C:\Users\Dani Marion>storj -u http://server015.storj.dk login
 [...]  > Enter your email address  >  meijesibbel@hotmail.com
 [...]  > Enter your password  >  *********

 [Sat Sep 10 2016 16:22:30 GMT+0200 (W. Europe Daylight Time)] [info]   This device has been successfully paired.

C:\Users\Dani Marion>storj -k meije0612 -u http://server015.storj.dk add-bucket Test 1000 1000
 [Sat Sep 10 2016 16:22:57 GMT+0200 (W. Europe Daylight Time)] [error]  Public key not registered

C:\Users\Dani Marion>

**Comment by MeijeSibbel** [comment]
Even if i delete the id_scdsa file and key.ring folder u still get the error that no public key is registered.
Registering with a different email also isn't working.

**Comment by MeijeSibbel** [comment]
This process works fine on the app.storj.io bridge btw.

**Comment by Th3Van** [comment]
hmm, there seems to be a problem with the -u / --url  option in storj.
You could try adding the server015.storj.dk manually.
Locate the index.js file like this `"/path/to/storj/lib/bridge-client/index.js"` (i guess your on windows, so i don't know wheres its default location is, since I'm using Debian)
Edit the file with notepad or wordpad and change this :
`baseURI: uri || process.env.STORJ_BRIDGE || 'https://api.storj.io',`
into 
`baseURI: uri || process.env.STORJ_BRIDGE || 'http://storj015.storj.dk',`

Save and try again without the `-u http://server015.storj.dk`  option.

**Comment by bookchin** [comment]
The -u option is borked right now, sorry. Set the environment variable `STORJ_BRIDGE` to the value you want.

**Comment by bookchin** [comment]
https://github.com/Storj/core/issues/420

**Comment by MeijeSibbel** [comment]
@Th3Van tried it but i get this:
```
C:\Users\Dani Marion>storj add-bucket Test 1000 1000
 [Sat Sep 10 2016 17:13:14 GMT+0200 (W. Europe Daylight Time)] [error]  getaddrinfo ENOTFOUND storj015.storj.dk storj015.storj.dk:80

C:\Users\Dani Marion>
```

**Comment by MeijeSibbel** [comment]
### i have it like this:
```
BridgeClient.prototype._checkOptions = function(uri, options) {
  options = merge({
    baseURI: uri || process.env.STORJ_BRIDGE || 'http://storj015.storj.dk',
    logger: new Logger(0),
    concurrency: 6,
    transferRetries: 3
  }, options);
```

**Comment by Th3Van** [comment]
woops,.. my bad, wrong server address.
Please change `storj015.storj.dk`  -> `server015.storj.dk` in the index.js file

**Comment by MeijeSibbel** [comment]
### Error
```
C:\Users\Dani Marion>storj add-bucket Files 1000 1000
 [Sat Sep 10 2016 17:18:47 GMT+0200 (W. Europe Daylight Time)] [error]  Invalid URI "server015.storj.dk/buckets"

C:\Users\Dani Marion>

```

**Comment by Th3Van** [comment]
Did you remember the `http://` like this : http://server015.storj.dk
`baseURI: uri || process.env.STORJ_BRIDGE || 'http://server015.storj.dk',`

I'm able to reproduce the error if i remove the `http://` in the index.js file
```
root@Server023:~/123# storj add-bucket test3333                                     
 [Sat Sep 10 2016 17:24:08 GMT+0200 (CEST)] [error]  Invalid URI "server023.storj.dk/buckets"
```

**Comment by MeijeSibbel** [comment]
It works!, i am now uploading to your bridge, thanks a buch for all the help. I will report if any error pop-up.

**Comment by MeijeSibbel** [comment]
The uploads are much more stable now, not a single bad-gateway error up till now, which is unique in my case as my instance would hit a bad-gateway error within the first 10 uploads, i have uploaded over 100 files now without a single issue, something is definitely wrong with the app.storj.io bridge.

**Comment by Th3Van** [comment]
I've added a DNS RR so that `server015` should be offloaded a bit :)
Instead of using `http://server015.storj.dk`, one should be using `http://apipool001.storj.dk` like this:
```
baseURI: uri || process.env.STORJ_BRIDGE || 'http://apipool001.storj.dk',
```
//Th3Van

**Comment by bookchin** [comment]
@th3van, no need to hard code anything, it's much easier to set the env var STORJ_BRIDGE.

**Comment by bookchin** [comment]
```
STORJ_BRIDGE=http://apipool001.storj.dk storj upload-file [...]
```

**Comment by vevarm** [comment]
The suggested fix in this thread worked for me as well, thanks.

**Comment by 1credit** [comment]
Oddly, when I just tried this, I received a "Bucket not found" right after the encryption step.

Copied the line in my script, commented it out, and removed the STORJ_BRIDGE prefix and the command runs normally (won't know for an hour if it finishes or not)

**Comment by bookchin** [comment]
@1credit the bucket wasn't found because its a different bridge. You have to register with the other bridge and create the bucket there.

**Comment by 1credit** [comment]
Gotcha.  Thanks.
